### ISSUE TITLE
Phoenix fails to launch after some time

### ISSUE THREAD

**Issue by metakirby5** [kasper/phoenix#167]
- Version: 2.5 (59)
- macOS: 10.12.3 (16D32)

Sometimes, after restarting Phoenix, it won't be responsive. Hovering over the icon in the menu bar results in the rainbow spinner. The following line is logged in Console:

```
default	16:44:45.507129 -0800	Phoenix	subsystem: com.apple.SkyLight, category: default, enable_level: 0, persist_level: 0, default_ttl: 7, info_ttl: 0, debug_ttl: 0, generate_symptoms: 0, enable_oversize: 0, privacy_setting: 0, enable_private_data: 0
```

Followed by a dozen or so of these lines:

```
default	16:44:45.514018 -0800	Phoenix	Local database thinks system build version is <private>, but the OS really is <private>.
```

I have to reboot in order for Phoenix to work properly again. Any idea what it could be?

**Comment by kasper** [comment]
I’m sorry to hear that! I haven’t noticed this behaviour, but it might be configuration related. However, those logs don’t seem to be related to Phoenix app. It’s rather possible that some Accessibility API is not responding in time and this is causing the hang. Unfortunately to debug, this would need more tighter reproduction steps.

**Comment by metakirby5** [comment]
It's quite random when this issue happens, so I'll post again if I find a way to consistently reproduce it.

**Comment by kasper** [comment]
Thanks!

**Comment by metakirby5** [comment]
I'm not sure under which conditions this happens, but I found today that whether or not Phoenix would successfully start depended on how long my config was.

This config works:

```coffee
#!/usr/bin/env coffee -p

# Cardinal directions
NORTH = 'north'
SOUTH = 'south'
EAST = 'east'
WEST = 'west'
DIR = 'DIR'
DIRS = [NORTH, SOUTH, EAST, WEST]

# Axes
VERTICAL = 'VERTICAL'
HORIZONTAL = 'HORIZONTAL'
AXIS = 'AXIS'
AXES = [VERTICAL, HORIZONTAL]

# Preferences
Phoenix.set openAtLogin: true
p =
  ## WINDOWS
  wins:
    tolerance: 10                # Tolerance for window movement (for pouring)
    unit: 100                    # Increment for moving and resizing
    gap: 20                      # Gap between windows
    key: 'val'
```

But this one does not:

```coffee
#!/usr/bin/env coffee -p

# Cardinal directions
NORTH = 'north'
SOUTH = 'south'
EAST = 'east'
WEST = 'west'
DIR = 'DIR'
DIRS = [NORTH, SOUTH, EAST, WEST]

# Axes
VERTICAL = 'VERTICAL'
HORIZONTAL = 'HORIZONTAL'
AXIS = 'AXIS'
AXES = [VERTICAL, HORIZONTAL]

# Preferences
Phoenix.set openAtLogin: true
p =
  ## WINDOWS
  wins:
    tolerance: 10                # Tolerance for window movement (for pouring)
    unit: 100                    # Increment for moving and resizing
    gap: 20                      # Gap between windows
    key: 'val'
    kkey: 'val'
```

Note the extra line: `kkey: 'val'`. Now that I can somewhat reproduce it, what do you suggest I do to get more information? I'm unfamiliar with macOS app debugging, but I know how to use things like `gdb`.

**Comment by metakirby5** [comment]
I managed to stumble my way through Xcode to the debugger, and found this:

![](https://ptpb.pw/cGDV.png)

It seems like Phoenix is hanging up when calling `PHShebangPreprocessor`. I've confirmed this theory; the issue does not recur after first compiling my config with `coffee -p`.

**Comment by metakirby5** [comment]
I think I fixed it :) Will send a PR shortly.

**Comment by kasper** [comment]
Brilliant! Nice work!

**Comment by kasper** [comment]
Fixed in #198.
